src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp

```diff
@@ -221,6 +237,9 @@ void scatter_elem_update_with_reduction(const DataType* input_data,
     }
 
     if (reduction_type == Reduction::MEAN) {
+        // this object will change the rounding mode only for integer types which is required to match torch
```

Comment:
why does it mention torch? Will we be aligned with other FWs?

Reply:
Based on @tomdol https://github.com/openvinotoolkit/openvino/pull/18069#discussion_r1236760148 from the core op PR:
> The `mean` reduction type is only present in pytorch and it always rounds down the calculated value to the nearest integer. We've added this reduction type to OV to cover this case in PT FE and I think it should be implemented accordingly.